File: chess/public/scripts/player.js
export default class Player {
  static counter = 0
  static domElement = [...document.getElementsByClassName('user-name')].reverse()
  constructor(name, side) {
    this.ID = ++Player.counter
    this.username = name
    this.side = side
    this.node = Player.domElement[side]
    this.node.textContent = name

    this.canCastle = true
    this.isInCheck = false
    this.kingReference = null
    this.pieces = {
      active: [],
      captured: []
    }
  }
}

File: java-games/CuriosChess/src/player/Player.java
package player;

/**
 * Player
 */
public class Player {

  public static byte playerCounter = 0;
  public String name = null;
  public final byte sideNumber = ++playerCounter;
  public int activePieces[];

  public Player(String _name) {
    name = _name;
  }
}